### DISCUSSION TITLE
Support for GemGIS Repository - Github Actions/Sphinx Docs/Conda recipe `t21-hack-gemgis`

### DISCUSSION THREAD

**Discussion by AlexanderJuestel** [softwareunderground/transform-2021-hackathon#4] [Projects]
## Slack channel
[t21-hack-gemgis](https://swung.slack.com/archives/C01SMN448KS)

## People
@AlexanderJuestel
@endarthur
@flohorovicic

## Description
This project is aiming at supporting the developers of the [GemGIS package](https://github.com/cgre-aachen/gemgis) with some expert knowledge about Github Actions, the Sphinx Documentation, releasing the package on the conda-force channel, Binder and other best practices for your open source project.

**What would be important to me (Alex) is that people who engage in this project not only complete tasks but also show the others how to do it and explain what they have been doing so we can pass that knowledge on to others!  Thanks!**

## Goals
* ~~Github Action Workflow to automatically run all the tests of the package when code is pushed or merged to master or the development branches~~
* Github Action Workflow to automatically build the Sphinx documentation when code is pushed or merged to the master branch
* ~~Github Action Workflow to automatically create a PyPi release/conda forge release when code is pushed or merged to the master branch~~
* ~~Creating a functioning `conda-forge recipe` for the GemGIS package - there is already a GemPy Recipe maintained by @AndrewAnnex which could be built on: https://github.com/conda-forge/gempy-feedstock/blob/master/recipe/meta.yaml / https://conda-forge.org/docs/maintainer/adding_pkgs.html~~
* Tips and tricks for the Sphinx documentation and setup of a `readthedocs.yaml` - @prisae
* Testing Binder integration for Tutorial Notebooks
* Tips and tricks with respect to dependencies and requirement files
* ~~Renaming master branch to main to be more inclusive! - https://github.com/github/renaming~~
* Maybe some brainstorming on the future course of the package and functionality that could be added
* Fixing Jupyter Notebooks 
* Draping geological maps over meshes - @endarthur 

## Resources

All discussion prior, during or after the Transform 2021 can be done in the GemGIS discussion section --> https://github.com/cgre-aachen/gemgis/discussions 

## Skills needed (if any)
If you're looking for teammates with particular skills, list them here.
* Geoscience
* Programming
* Testing
* Documentation

**Comment by AlexanderJuestel** [comment]
https://conda-forge.org/docs/maintainer/adding_pkgs.html

**Comment by endarthur** [comment]
Some ideas for GemGIS development during this transform:
1. Interactive interface for GemGIS (eg drawing sections and such)
2. smoother maps, maybe directly using the interpolator in grid-free contouring
3. More IO formats
4. Better integration with QGIS (QML and projects) and maybe ArcGIS?
5. Fun with AR and VR?
6. Cookie cutter templates for GemGIS projects (would that be interesting or useful in any sense? Something like building a folder and file structure for a new project, maybe even based on a QGIS project file or something similar?)

**Comment by filippo82** [comment]
@AlexanderJuestel let me know if AWS could be of any help in this project 💪🏻

**Reply by AlexanderJuestel** [reply]
Sure, I have never worked with AWS before or used the services so I am keen to explore it if you could elaborate a little more on it :)

**Reply by filippo82** [reply]
eheh, I really cannot say much more than that we have plenty of credits to use :D

**Reply by filippo82** [reply]
So even for the CI/CD part, if one of their services could be of use, we can set you up with an IAM account to test it.

**Reply by AlexanderJuestel** [reply]
Okay, then I have to figure out what these services are and of what use they could be. Maybe @flohorovicic has some experience with that as well ;)

**Reply by RichardScottOZ** [reply]
I can use credits pretty fast ;)

**Comment by flohorovicic** [comment]
...and (for me): _finally_ adjusting the import/exports for GoogleEarth.

**Comment by RichardScottOZ** [comment]
@endarthur that all sounds good, I just want automagic 3d density block models currently

**Reply by RichardScottOZ** [reply]
So possible case study example from messy large scale stuff through here.

**Comment by filippo82** [comment]
[T21 Hackathon Idea Mashup](https://softwareunderground.org/events/2021/4/7/t21-hackathon-idea-mashup) today (or tomorrow depending on your time zone) Wednesday April 14th at 16:00 UTC. Come to Slack to find the link for the meeting!
